src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,114 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        sizeCheck();
+        int hash = hashKey(key);
+        int bucket = hash % capacity;
+        if (!overwriteDuplicates(bucket, hash, key, value)) {
+            Node<K, V> newNode = new Node<>(hash, key, value, null);
+            if (table[bucket] == null) {
+                table[bucket] = newNode;
+                size++;
+            } else {
```

Comment:
let's move this code (28-31 lines) before 
`if (!overwriteDuplicates(bucket, hash, key, value)) {`
and `Node<K, V> newNode = new Node<>(hash, key, value, null);` too